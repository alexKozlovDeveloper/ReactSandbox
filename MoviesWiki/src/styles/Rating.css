.title {
    border: 2px solid #BADA55;
    padding: 6px;
    margin: 2px;
    border-radius: 50%;
    background-color: #222327;
    color: gold;
}