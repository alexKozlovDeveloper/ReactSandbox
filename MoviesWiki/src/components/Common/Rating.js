import React, { Component, useMemo } from "react";

import styles from "../../styles/Rating.css"

const Rating = ({ rating }) => {
    return useMemo(()=> {
        return  <div className={styles.title}>
                    {rating}
                </div>
    });
}

export default Rating;