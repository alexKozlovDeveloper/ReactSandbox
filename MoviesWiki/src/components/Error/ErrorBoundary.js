import React, { Component } from 'react';

import styles from '../../styles/ErrorBoundary.css';

class ErrorBoundary extends Component {
  constructor(props) {
    super(props);
    this.state = { hasError: false };
  }

  static getDerivedStateFromError(error) {
    // Обновить состояние с тем, чтобы следующий рендер показал запасной UI.
    console.log(error);
    return { hasError: true };
  }

  // componentDidCatch(error, errorInfo) {
  //   // Можно также сохранить информацию об ошибке в соответствующую службу журнала ошибок
  //   // logErrorToMyService(error, errorInfo);
  //   // console.log(error);
  // }

  render() {
    if (this.state.hasError) {
      // Можно отрендерить запасной UI произвольного вида
      return <div className={styles.container}>Oops... something went wrong!</div>;
    }

    return this.props.children;
  }
}

export default ErrorBoundary;
