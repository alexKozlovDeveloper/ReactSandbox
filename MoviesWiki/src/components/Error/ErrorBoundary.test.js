import React from 'react';
import sinon from 'sinon';
import { shallow, mount, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import ErrorBoundary from './ErrorBoundary';
import Rating from '../Common/Rating';

configure({ adapter: new Adapter() });

describe('ErrorBoundary component', () => {
  it('Rendering test', () => {
    const component = mount(<ErrorBoundary>
            <div>ak test</div>
        </ErrorBoundary>);
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('Rendering test with error', () => {
    const spy = sinon.spy();

    const component = shallow(<ErrorBoundary spy={spy}><Rating rating="9"/></ErrorBoundary>);

    const error = new Error('Test Error!');

    component.find(Rating).simulateError(error);

    expect(component.state().hasError === true).toBe(true);
  });
});
