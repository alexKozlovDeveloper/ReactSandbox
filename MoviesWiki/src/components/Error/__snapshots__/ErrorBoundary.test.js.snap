// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary component Rendering test 1`] = `
<ErrorBoundary>
  <div>
    ak test
  </div>
</ErrorBoundary>
`;
